### ISSUE TITLE
[CMake] Use the default state icons as fallback if they're not present in OEM_THEME_DIR

### ISSUE THREAD

**Issue by SamuAlfageme** [owncloud/client#6296]
I guess this is a regression since I never actually used a replacement for the state icons in https://github.com/owncloud/client/tree/master/theme/colored and my themed builds w/ `OEM_THEME_DIR`didn't fail. 

```
make[3]: *** No rule to make target `/usr/client/mirall/theme/colored/state-error-16.png', needed by `src/gui/qrc_theme.cpp'.  Stop.
make[2]: *** [src/gui/CMakeFiles/theme_test_autogen.dir/all] Error 2
make[1]: *** [src/gui/CMakeFiles/theme_test.dir/rule] Error 2
make: *** [theme_test] Error 2
```

cc/ @ogoffart @dschmidt

**Comment by ckamm** [comment]
@SamuAlfageme How do you trigger this error? I haven't seen it. Do you have `OEM_THEME_DIR` set?

**Comment by SamuAlfageme** [comment]
@ckamm yup - I was compiling on macOS tho. Didn't check other platforms. 

My CMake flags:

```
-DOPENSSL_ROOT_DIR=/usr/local/opt/openssl
-DOPENSSL_INCLUDE_DIR=/usr/local/opt/openssl/include
-DOPENSSL_LIBRARIES=/usr/local/opt/openssl/lib
-DCMAKE_PREFIX_PATH=/usr/local/opt/qt5/lib/cmake
-DWITH_DOC=FALSE
-DOEM_THEME_DIR=/Users/salfageme/Desktop/mirall
```

And the theme:

```
➜  ~ tree Desktop/mirall
Desktop/mirall
├── OEM.cmake
├── admin-theme.h
├── desktop.properties.xml
├── installer
│   └── nsi
│       ├── page_header.bmp
│       └── welcome.bmp
├── package.cfg
├── theme
│   └── colored
│       ├── whatever-icon-128.png
│       ├── whatever-icon-32.png
│       ├── whatever-icon-48.png
│       ├── whatever-icon-64.png
│       ├── wizard_logo.png
│       └── wizard_logo@2x.png
└── theme.qrc
```

**Comment by jnweiger** [comment]
@SamuAlfageme  is there a branding that reproduces this issue in jenkins.int ?

**Comment by SamuAlfageme** [comment]
@jnweiger check `alfageme/mirall` saw this one precisely with it. At first blamed the GitHub -> Gitea migration, but generated a new theme with ownBrander locally and when not selecting "custom state icons" still wasn't using the default ones.

**Comment by guruz** [comment]
I moved this out of 2.4.1 because i don't understand how this can be a regression.
The icons are specified in theme.qrc, I don't think there can be some conditional code in there to take some files from original some from theme?

Is this related to 9a021d227929fbe939d19e0fef16401adbde877c ?

**Comment by ckamm** [comment]
@SamuAlfageme Is this still a problem? How can we reproduce it?

**Comment by ckamm** [comment]
I'll try to reproduce like @SamuAlfageme explained above.

**Comment by ckamm** [comment]
@dschmidt When trying to use @SamuAlfageme's theme I noticed that the expected application icon file name changed in fe3c68b9dd64cbb391421e2f444aca27e72a6937 but I don't see a corresponding change in ownbrander/master. Maybe his theme doesn't work because one needs to use a different ownbrander version with 2.5?

**Comment by ckamm** [comment]
To summarize: it looks like there have been changes around how themes are structured and maybe Samu's issue was just because an older theme was used with a newer client version.

**Comment by ogoffart** [comment]
@SamuAlfageme is there still an issue? Or can it be closed?  Were you using an outdated theme?

**Comment by SamuAlfageme** [comment]
@ogoffart @ckamm will generate a theme with ownBrander from scratch and verify if it works.

**Comment by ogoffart** [comment]
Closing old issue.

**Comment by ogoffart** [comment]
Since the issue was about a potential behavior change in the theming from 2.4 to 2.5, that would have had to be fixed in 2.5.0. Even if there was indeed a difference, we can't change it back in 2.6
